File: 1. Two Sum/1.js
// 1. Two Sum solution in JavaScript
// https://leetcode.com/problems/two-sum/description/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let map = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        if (map.get(nums[i]) === undefined) {
            map.set(target - nums[i], i);
        } else {
            return [map.get(nums[i]), i];
        }
    }
};

File: 1. Two Sum/1.go
// 1. Two Sum solution in GoLang
// https://leetcode.com/problems/two-sum/description/

package leetcode

func twoSum(nums []int, target int) []int {
    index_map := make(map[int]int)

    for i := 0; i < len(nums); i++ {
        value, is_present := index_map[nums[i]]
        if  is_present == false {
            index_map[target - nums[i]] = i
        } else {
            return []int{value, i}
        }
    }

    return []int{}
}